src/wp-includes/class-wp-query.php

```diff
@@ -4856,6 +4856,21 @@ protected function generate_cache_key( array $args, $sql ) {
 			$args['suppress_filters']
 		);
 
+		if ( empty( $args['post_type'] ) ) {
+			if ( $this->is_attachment ) {
+				$args['post_type'] = 'attachment';
+			} elseif ( $this->is_page ) {
+				$args['post_type'] = 'page';
+			} else {
+				$args['post_type'] = 'post';
+			}
+		}
+		$args['post_type'] = (array) $args['post_type'];
+
+		if ( ! isset( $q['orderby'] ) ) {
+			$args['orderby'] = 'date';
+		}
```

Comment:
This looks right, based on [this block](https://github.com/WordPress/wordpress-develop/blob/3b9aefb06cae1f34ed19b08caf3b3ed32df04b73/src/wp-includes/class-wp-query.php#L2432-L2442). An inline comment would be helpful here.

It makes me wonder if we should actually remove the `orderby` and just use the ORDER BY clause from the `$sql` here instead, as there are other places where differing `$q['orderby']` params get converted into the same query. I don't see that as a blocker, however.

Reply:
Added.